### PR TITLE
Feature/filter-Doctors

### PR DESCRIPTION
**Pull request by zarmeenmustafa651**

### CODE REVIEW

File: app/controllers/doctors_controller.rb [hunk 1]
```diff
@@ -107,4 +107,10 @@ def show_page_breadcrumb
     add_breadcrumb t('doctor.breadcrumb.show'), doctor_path
   end
 
+  # POST /show_availabilities
```
Comment (itratzahra7) [inline app/controllers/doctors_controller.rb]: change this

File: app/helpers/doctor_helper.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module DoctorHelper
+  def list_of_doctors_with_speciality(doctors, speciality)
+    if speciality.present?
```
Comment (itratzahra7) [inline app/helpers/doctor_helper.rb]: tertiary

File: app/views/doctors/_table.html.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    <tr>
+      <th><%= t('doctor.label.picture') %></th>
+      <th><%= t('doctor.label.doctor_name') %></th>   
+      <th><%= t('doctor.label.registration_no') %></th>   
+      <th><%= t('doctor.label.speciality') %></th>   
+      <th><%= t('doctor.label.consultancy_fee') %></th>  
+    </tr>   
+  </thead>   
+  <tbody>   
+  <% list_of_doctors_with_speciality(@doctors, params[:Filter_Doctor]).each do |doctor| %>
```
Comment (MirahSaqib) [inline app/views/doctors/_table.html.erb]: params[:Filter_Doctor] -> params[:filter]

File: app/views/doctors/_table.html.erb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      <th><%= t('doctor.label.registration_no') %></th>   
+      <th><%= t('doctor.label.speciality') %></th>   
+      <th><%= t('doctor.label.consultancy_fee') %></th>  
+    </tr>   
+  </thead>   
+  <tbody>   
+  <% list_of_doctors_with_speciality(@doctors, params[:Filter_Doctor]).each do |doctor| %>   
+    <tr>
+      <td>
+        <a href=<%= doctor_path(doctor) %> >
```
Comment (MirahSaqib) [inline app/views/doctors/_table.html.erb]: use link_to

File: app/views/doctors/_table.html.erb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        <div class="set_image">
+          <% if doctor.profile_picture.present? %>
+            <%= image_tag doctor.profile_picture.url, class: 'block' %>
```
Comment (MirahSaqib) [inline app/views/doctors/_table.html.erb]: image_pack_tag ?

File: config/routes.rb [hunk 6]
```diff
@@ -59,6 +59,9 @@
     end
     resources :doctors do
       resources :availabilities, except: [:edit, :update, :show]
+      collection do
+        post 'speciality_filter'
```
Comment (MirahSaqib) [inline config/routes.rb]: why post?

File: app/views/doctors/_speciality_filter.html.erb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<%= form_tag('/doctors/speciality_filter', remote: true) do %>
```
Comment (MirahSaqib) [inline app/views/doctors/_speciality_filter.html.erb]: doctor_speciality_filter_path

File: app/views/doctors/_speciality_filter.html.erb [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+  <div class="row">
+    <%= select_tag 'Filter_Doctor', options_for_select(@doctors.map { |doctor| doctor.speciality}.uniq), {include_blank: 'Select Speciality', class: 'speciality-filter form-control col-2 ml-4'} %>
```
Comment (MirahSaqib) [inline app/views/doctors/_speciality_filter.html.erb]: Filter_Doctor -> filter

File: app/controllers/doctors_controller.rb [hunk 9]
```diff
@@ -107,4 +107,10 @@ def show_page_breadcrumb
     add_breadcrumb t('doctor.breadcrumb.show'), doctor_path
   end
 
+  # POST /speciality_filter
+  def speciality_filter
+    respond_to do |format|
```
Comment (MirahSaqib) [inline app/controllers/doctors_controller.rb]:
```
@doctors = @doctors.where(speciality: params[:filter]) if params[:filter].present?

```

File: app/helpers/doctor_helper.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+module DoctorHelper
```
Comment (MirahSaqib) [inline app/helpers/doctor_helper.rb]: remove file

File: app/views/doctors/speciality_filter.js.erb [hunk 11]
```diff
@@ -0,0 +1 @@
+$(".table").html("<%= escape_javascript(render partial: 'doctors/table') %>");
```
Comment (MirahSaqib) [inline app/views/doctors/speciality_filter.js.erb:1]: pass locals

``` $(".table").html("<%= escape_javascript(render(partial: 'doctors/table', locals: { doctors: @doctors })) %>");```

File: app/views/doctors/index.html.erb [hunk 12]
```diff
@@ -11,38 +11,8 @@
...
-            <td ><%= link_to doctor.name, doctor_path(doctor) %></td>   
-            <td><%= doctor.registration_no %></td>  
-            <td><%= doctor.speciality %></td>  
-            <td><%= doctor.consultancy_fee %></td>
-          </tr>   
-        <% end %>  
-        </tbody>
-      </table> 
+      <%= render partial: 'speciality_filter' %>
+      <%= render partial: 'table' %>
```
Comment (MirahSaqib) [inline app/views/doctors/index.html.erb:15]: ``` <%= render(partial: 'table', locals: { doctors: @doctors }) %> ```

File: app/views/doctors/_table.html.erb (hunk 3 above)
Comment (MirahSaqib) [inline app/views/doctors/_table.html.erb]: use doctors passed in locals directly

```doctors.each do |doctor|```

**Pull request merged**